sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationEditFragment.java

```diff
@@ -0,0 +1,220 @@
+		contentBinding.immunizationStartDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationEndDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationLastInfectionDate.initializeDateField(getFragmentManager());
+
+		contentBinding.immunizationMeansOfImmunization.addValueChangedListener(e -> {
+			if (e.getValue() == MeansOfImmunization.OTHER || e.getValue() == MeansOfImmunization.RECOVERY) {
+				contentBinding.immunizationImmunizationManagementStatus.setValue(ImmunizationManagementStatus.COMPLETED);
+				contentBinding.immunizationImmunizationManagementStatus.setEnabled(false);
+				contentBinding.overwriteImmunizationManagementStatusCheckBox.setVisibility(View.VISIBLE);
+			}
+			if (e.getValue() == MeansOfImmunization.VACCINATION || e.getValue() == MeansOfImmunization.VACCINATION_RECOVERY) {
+				contentBinding.immunizationVaccinationHeading.setVisibility(View.VISIBLE);
+				contentBinding.immunizationRecoveryHeading.setVisibility(View.VISIBLE);
+
+				contentBinding.immunizationNumberOfDoses.setVisibility(View.VISIBLE);
+				contentBinding.immunizationRecoveryDate.setVisibility(View.VISIBLE);
+				contentBinding.immunizationPositiveTestResultDate.setVisibility(View.VISIBLE);
+
+				contentBinding.immunizationNumberOfDoses.setEnabled(true);
+				contentBinding.immunizationRecoveryDate.setEnabled(true);
+				contentBinding.immunizationPositiveTestResultDate.setEnabled(true);
+			} else {
+				contentBinding.immunizationVaccinationHeading.setVisibility(View.GONE);
+				contentBinding.immunizationRecoveryHeading.setVisibility(View.GONE);
+				contentBinding.immunizationNumberOfDoses.setVisibility(View.GONE);
+				contentBinding.immunizationRecoveryDate.setVisibility(View.GONE);
+				contentBinding.immunizationPositiveTestResultDate.setVisibility(View.GONE);
+			}
+		});
+
+		contentBinding.overwriteImmunizationManagementStatusCheckBox.addValueChangedListener(e -> {
+			if (Boolean.TRUE.equals(e.getValue())) {
+				contentBinding.immunizationImmunizationManagementStatus.setEnabled(true);
+			} else {
+				contentBinding.immunizationImmunizationManagementStatus.setEnabled(false);
+			}
+		});
+
+		contentBinding.immunizationImmunizationManagementStatus.addValueChangedListener(e -> {
+			if (e.getValue() == ImmunizationManagementStatus.SCHEDULED || e.getValue() == ImmunizationManagementStatus.ONGOING) {
+				contentBinding.immunizationImmunizationStatus.setValue(ImmunizationStatus.PENDING);
+			}
+			if (e.getValue() == ImmunizationManagementStatus.COMPLETED) {
+				contentBinding.immunizationImmunizationStatus.setValue(ImmunizationStatus.ACQUIRED);
+			}
+			if (e.getValue() == ImmunizationManagementStatus.CANCELED) {
+				contentBinding.immunizationImmunizationStatus.setValue(ImmunizationStatus.NOT_ACQUIRED);
+			}
+		});
+
+		contentBinding.immunizationImmunizationManagementStatus.setValue(ImmunizationManagementStatus.SCHEDULED);
```

Comment:
This should only be done in the NewFragment (because the edit fragment shouldn't need a default)